sourcemash/frontend/static/js/views/sidenav_view.js

```diff
@@ -6,8 +6,17 @@ Sourcemash.Views.SidenavView = Backbone.View.extend({
     this.feeds = options.feeds;
     this.categories = options.categories;
     this.loading = true;
+    this.feedsChanged = false;
+
     this.listenTo(this.feeds, 'sync change:unread', this.render);
-    this.listenTo(this.feeds, 'change:subscribed', function(){ this.categories.fetch(); });
+    this.listenTo(this.feeds, 'change:subscribed',
+      function(){
+        // Don't tell the user to refresh categories until fetch is complete
+        if (!this.loading) {
```

Comment:
Yes. First, because we don't want both a refresh button and a loading wheel appearing simultaneously. It looks like we don't know what to tell the user. It should be "act" *or* "wait", not act *and* wait.